tensorflow/core/common_runtime/dml/dml_tracing.h

```diff
@@ -5,18 +5,20 @@
 // Helper for adding tracing events useful in ETW-based perf analysis
 // (GPUView/WPA). No telemetry.
 class DmlTracing {
+ public:
+  // GUID to associate a PIX event name with an IDMLObject (there is an existing
+  // SetName method, but no public GetName method).
+  constexpr static GUID kPixEventNameId = {0x191d7d5, 0x9fe1, 0x47cc, 0xb6,
+                                           0x46,      0xf6,   0x78,   0xb6,
+                                           0x33,      0x2f,   0x96};
+
```

Comment:
>GUID to associate ... with an IDMLObject
Is this GUID expected to match the GUID of IDMLObject? (which is "c8263aac-9e0c-4a2d-9b8e-007521a3317c")

Reply:
Nope, just a random GUID only used by TFDML. [IDMLObject](https://docs.microsoft.com/en-us/windows/win32/api/directml/nn-directml-idmlobject) (similar to other base DX-style interfaces) supports a few methods for attaching application-defined data to an instance of the interface. This is sometimes convenient since it allows us to stuff arbitrary data into the operator, like a name for the PIX event, without having to store it separately and pass it through each layer of the DML backend as a separate parameter.

Normally we'd call [IDMLObject::SetName](https://docs.microsoft.com/en-us/windows/win32/api/directml/nf-directml-idmlobject-setname) to attach a string/name to the object, which can be used in PIX events that DirectML itself emits, but unfortunately there's no public method to get the name back out of the object.

Reply:
Should we add one?